src/main/java/org/opensearch/security/privileges/ActionPrivileges.java

```diff
@@ -901,11 +1022,69 @@ PrivilegesEvaluatorResponse providesExplicitPrivilege(
                     }
                 }
             }
-
             return PrivilegesEvaluatorResponse.insufficient(checkTable)
                 .reason("No explicit privileges have been provided for the referenced indices.")
                 .evaluationExceptions(exceptions);
         }
+
+        PrivilegesEvaluatorResponse apiTokenProvidesIndexPrivilege(
+            PermissionBasedPrivilegesEvaluationContext context,
+            IndexResolverReplacer.Resolved resolvedIndices,
+            Set<String> actions,
+            Boolean explicit
+        ) {
+            Permissions permissions = context.getPermissions();
+            List<ApiToken.IndexPermission> indexPermissions = permissions.getIndexPermission();
+
+            for (String concreteIndex : resolvedIndices.getAllIndices()) {
+                boolean indexHasAllPermissions = false;
+
+                // Check each index permission
+                for (ApiToken.IndexPermission indexPermission : indexPermissions) {
+                    // First check if this permission applies to this index
+                    boolean indexMatched = false;
+                    for (String pattern : indexPermission.getIndexPatterns()) {
+                        if (WildcardMatcher.from(pattern).test(concreteIndex)) {
```

Comment:
OpenSearch has the property that privileges on aliases also grant privileges on indices contained inside the alias. Similar properties hold for data streams and backing indices. I think this support is missing here.